src/components/Api.js

```diff
@@ -0,0 +1,96 @@
+export default class Api { 
+  constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+  }
+
+  _checResponse(res) {
+      if (res.ok) {
+          return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+  }
+
+  _request(name, link) {

```

Comment:
`name, link` should be named `url, options` because you pass an `url` and an object of options


```js
  _request(url, options) {
      return fetch(url, options).then(this._checResponse);
  }

```